File: first/php/library/person.php
<?php
     abstract class Person {
        private int $identification; 
        private ?string $name;

        public function __construct(int $identification, ?string $name) {
            $this->name = $name;
            $this->identification = $identification;
        }
        public function  getName(){
            return $this->name;
        } 
        public function  getIdentification(){
            return $this->identification;
        } 
        abstract public function calculateSalary() : int;
        
    }?>

File: first/java/Person.java

public abstract class Person {
    private final int identification; 
    private final String name;        

    public Person(int identification, String name) {
        this.identification = identification;
        this.name = name;
    }

    public String getName() {
        return this.name;
    }

    public int getIdentification() {
        return this.identification;
    }

    public abstract int calculateSalary();
}